proposals/000-ensure-compilation-scalability.md

```diff
@@ -1,56 +1,99 @@
-# SCP-[000]: Improve user experience for builds that use only direct dependencies
+# SCP-[000]: Minimization of compilation classpaths
 
 ## Proposer
 
 Proposed by Stu Hood, Eugene Burmako - Twitter - November 2016
 
 ## Abstract
 
-The most commonly discussed angle of attack for improving the Scala build experience is improving the speed of the compiler. Instead, this document addresses a different angle: improving the experience of using only "direct" dependencies in builds, in order to improve cache hit rates and unblock scaling multiple remote invocations of the compiler.
+The most commonly discussed angle of attack for improving the Scala build experience is speeding up the compiler. Instead, this proposal addresses a different angle: minimizing compilation classpaths to enable distributed builds.
 
-## Background
+## Minimization of compilation classpaths
 
-A "direct" (also known as "strict") dependency is defined here as a dependency that is specifically mentioned in the build definition for some compilation unit. "direct" dependencies are thus differentiated from "transitive" dependencies: those that are declared by collecting the transitive closure of the direct dependencies of some compilation unit.
+Our build workflow at Twitter is centered around a monorepo. Instead of keeping internal projects in separate repositories, we merge them into one repository (hence the name “monorepo”), significantly reducing the overhead associated with evolving independent repositories.
 
-When building very large Scala projects, the presence/necessity of transitive build dependencies on the compile classpath represents a very real limit on the ability to distribute compilation: shipping 2000 transitive dependencies to a cluster of remote compiler instances is significantly more challenging than shipping 20 direct dependencies. Additionally, over-declaring dependencies causes cache keys for build artifacts to be invalidated more often than is strictly necessary.
+In comparison with a more common workflow that involves a number of coarse-grained projects and intra-project incremental compilation via sbt, our build workflow relies on many more (tens of thousands) fine-grained modules and inter-project incrementality. This workflow is enabled by Pants, our open-source build tool that understands a multitude of languages including Java, Scala, Python and others.
```

Comment:
This section should mention that this is the default behaviour buck and bazel as well, simply to highlight the fact that this is a common approach for large projects.